stablehlo/dialect/StablehloOps.td

```diff
@@ -1448,7 +1447,7 @@ def StableHLO_AllToAllOp : StableHLO_Op<"all_to_all",
 def StableHLO_ReduceOp: StableHLO_ShapedInterfaceOp<"reduce", [
       RecursiveMemoryEffects,
       SameVariadicOperandSize /*reduce_c3*/,
-      InferTensorTypeWithReify /*reduce_c7*/,
+      InferTensorTypeWithReify /*reduce_c7, reduce_c8*/,
       SingleBlockImplicitTerminator<"ReturnOp">
     ]> { /*reduce_c7*/
   let summary = "Reduce operation";
```

Comment:
Can't leave comment below this line, but ~15 lines down theres a TODO comment that I think we can delete now:

```
  // TODO(hinsu): Verify that the attached body arguments and results are
  // compatible with reduce op's operands.
```